From 57b1b34e76f9f5d548dca12140abe3a154d913ba Mon Sep 17 00:00:00 2001
From: Dezhi Kong <dezhi.kong@amlogic.com>
Date: Thu, 16 Jul 2015 15:20:10 +0800
Subject: [PATCH 5763/5965] PD#108700: dts: g9tv/g9bb: use CMA method in tvin
 module

Update for the omited dts of n304 and n210.

This commit is a patch for commit:
a3ea12a1e2579e99876eb8fb60275a4d5fe36bee

Owner: dezhi.kong@amlogic.com

Change-Id: I10271034cad33d5dd16a5461f2a5776308d13c6a
---
 arch/arm/boot/dts/amlogic/mesong9bb_n304.dtd | 47 ++++++------
 arch/arm/boot/dts/amlogic/mesong9tv_n210.dtd | 76 ++++++++++++++------
 2 files changed, 77 insertions(+), 46 deletions(-)

diff --git a/arch/arm/boot/dts/amlogic/mesong9bb_n304.dtd b/arch/arm/boot/dts/amlogic/mesong9bb_n304.dtd
index 3ef08b01c116..1872c1e696c9 100644
--- a/arch/arm/boot/dts/amlogic/mesong9bb_n304.dtd
+++ b/arch/arm/boot/dts/amlogic/mesong9bb_n304.dtd
@@ -82,38 +82,37 @@
 //$$ L2 PROP_U32 = "aml_reserved_end"
 //$$ L2 PROP_U32 6 = "linux,usable-memory"
 	memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
 		device_type = "memory";
 		aml_reserved_start = <0x06000000>; /**reserved memory start */
-		aml_reserved_end = <0x05000000>;/**reserved memory end : dtb start for uboot*/
+		aml_reserved_end = <0x05000000>; /**reserved memory end : dtb start for uboot*/
 		phys_offset = <0x00200000>;
 		linux,total-memory = <0x3fe00000>; // 1G reference m8b
-        #address-cells = <1>;
-        #size-cells = <1>;
-
-        cma_0:region@0 {
-           region_name = "cma_0";
-           reg = <0 0x01200000>;
-           linux,contiguous-region;
-        };
 
-        cma_vdin0: region@2 {
-		region_name = "cma_vdin0";
-		linux,contiguous-region;
-		reg = <0x00000000 0x02100000>;
-	};
+		cma_0:region@0 {
+			region_name = "cma_0";
+			reg = <0 0x01200000>;
+			linux,contiguous-region;
+		};
 
-	cma_vdin1: region@3 {
-		region_name = "cma_vdin1";
-		linux,contiguous-region;
-		reg = <0x00000000 0x01000000>;
-	};
+		cma_vdin0: region@2 {
+			region_name = "cma_vdin0";
+			linux,contiguous-region;
+			reg = <0x00000000 0x02100000>;
+		};
 
-	cma_tvafe: region@4 {
-		region_name = "cma_tvafe";
-		linux,contiguous-region;
-		reg = <0x00000000 0x00500000>;
-	};
+		cma_vdin1: region@3 {
+			region_name = "cma_vdin1";
+			linux,contiguous-region;
+			reg = <0x00000000 0x01000000>;
+		};
 
+		cma_tvafe: region@4 {
+			region_name = "cma_tvafe";
+			linux,contiguous-region;
+			reg = <0x00000000 0x00500000>;
+		};
 	};
 
 
diff --git a/arch/arm/boot/dts/amlogic/mesong9tv_n210.dtd b/arch/arm/boot/dts/amlogic/mesong9tv_n210.dtd
index bc190e7bce13..bf2df120d79f 100644
--- a/arch/arm/boot/dts/amlogic/mesong9tv_n210.dtd
+++ b/arch/arm/boot/dts/amlogic/mesong9tv_n210.dtd
@@ -95,14 +95,32 @@
 		mem_4k2k_decoder: region@1 {
 			region_name = "cma_4k2k";
 			linux,contiguous-region;
-			reg = <0x00000000 0x0c000000>;
+			reg = <0x00000000 0x0ca00000>;
 		};
 
 		cma_0:region@0 {
 			region_name = "cma_0";
-			reg = <0 0x01200000>;
+           reg = <0 0x02400000>;
 			linux,contiguous-region;
 		};
+
+		cma_vdin0: region@2 {
+			region_name = "cma_vdin0";
+			linux,contiguous-region;
+			reg = <0x00000000 0x04000000>;
+		};
+
+		cma_vdin1: region@3 {
+			region_name = "cma_vdin1";
+			linux,contiguous-region;
+			reg = <0x00000000 0x01000000>;
+		};
+
+		cma_tvafe: region@4 {
+			region_name = "cma_tvafe";
+			linux,contiguous-region;
+			reg = <0x00000000 0x00500000>;
+		};
 	};
 
 
@@ -811,8 +829,7 @@ NON sdio device(means sd/mmc card), other:reserved */
 		compatible = "amlogic,ppmgr";
 		dev_name = "ppmgr";
 		status = "okay";
-		reserve-memory = <0x01000000>; // 16M
-//		reserve-iomap = "true";
+		reserve-memory = <0x02000000>; // 32M
 	};
 
 
@@ -861,13 +878,15 @@ NON sdio device(means sd/mmc card), other:reserved */
 		dev_name = "vdin0";
 		status = "ok";
 		//reserve-memory = <0x01000000>;
-		reserve-memory = <0x04000000>;
-		reserve-iomap = "true";
+		//reserve-memory = <0x04000000>;
+		//reserve-iomap = "true";
 		//max_size = <1920 1080>;//1920x1080x2x4=17M
-		max_size = <4096 2160>;//4096x2160x2x4=67M
+		//max_size = <4096 2160>;//4096x2160x2x4=67M
 
 		irq = <115>;
 		vdin_id = <0>;
+		max_size = <64>;//64M
+		linux,contiguous-region = <&cma_vdin0>;
 	};
 
 // DEVICE="vdin1"
@@ -879,11 +898,13 @@ NON sdio device(means sd/mmc card), other:reserved */
 		compatible = "amlogic,vdin";
 		dev_name = "vdin1";
 		status = "ok";
-		reserve-memory = <0x01000000>;
-		reserve-iomap = "true";
-		miracast_size = <1920 1080>;//1920x1080x2x4=17M
+		//reserve-memory = <0x01000000>;
+		//reserve-iomap = "true";
+		//miracast_size = <1920 1080>;//1920x1080x2x4=17M
 		irq = <117>;
 		vdin_id = <1>;
+		max_size = <16>;//16M
+		linux,contiguous-region = <&cma_vdin1>;
 	};
 
 
@@ -891,8 +912,8 @@ NON sdio device(means sd/mmc card), other:reserved */
 		compatible = "amlogic,tvafe";
 		dev_name = "tvafe";
 		status = "ok";
-		reserve-memory = <0x500000>;
-		reserve-iomap = "true";
+		//reserve-memory = <0x500000>;
+		//reserve-iomap = "true";
 		tvafe_id = <0>;
 		pinctrl-names = "default";
 		tvafe_pin_mux = < /*!!particular sequence, no more and no less!!!*/
@@ -901,8 +922,20 @@ NON sdio device(means sd/mmc card), other:reserved */
 			2	/*// TVAFE_CVBS_IN1,  //CVBS_IN2*/
 			4	/*// TVAFE_CVBS_IN3,    //CVBS_IN3,*/
 		>;
+		max_size = <5>;//5M
+		linux,contiguous-region = <&cma_tvafe>;
+	};
+
+	amlvecm {
+		compatible = "amlogic,vecm";
+		dev_name = "aml_vecm";
+		status = "okay";
+		gamma_en = <1>;/*  1:enabel ;0:disable*/
+		wb_en = <1>;/*  1:enabel ;0:disable*/
+		cm_en = <1>;/*  1:enabel ;0:disable*/
 	};
 
+
 /// ***************************************************************************************
 ///	-	DISP&MM-AMLVIDEO2
 //$$ MODULE = "DISP&MM-AMLVIDEO2"
@@ -917,21 +950,20 @@ NON sdio device(means sd/mmc card), other:reserved */
         reserve-iomap = "true";
     };
 
+
 /// ****************************************************************************
 ///	-	ION
 //$$ MODULE="ION"
 //$$ DEVICE="ion_dev"
 //$$ L2 PROP_STR = "status"
-	ion_dev {
-		compatible = "amlogic,ion_dev";
-		dev_name = "ion_dev";
-		status = "ok";
-		reserve-memory = <0x01000000>; //16M
-//		reserve-iomap = "true";
-	};
-
-
-
+    ion_dev{
+        compatible = "amlogic,ion_dev";
+        dev_name = "ion_dev";
+        status = "ok";
+        share-memory-name = "ppmgr0";
+        share-memory-offset = <0>;
+        share-memory-size = <0x02000000>; //32M
+    };
 
 /// ****************************************************************************
 ///	-	DISP&MM-Vout
-- 
2.19.0

